src/vcpkg/postbuildlint.cpp

```diff
@@ -290,11 +289,12 @@ namespace vcpkg::PostBuildLint
         std::vector<path> misplaced_cmake_files;
         for (auto&& dir : dirs)
         {
-            auto files = fs.get_files_recursive(dir);
-            for (auto&& file : files)
+            for (auto&& file : fs.get_regular_files_recursive(dir, IgnoreErrors{}))
             {
-                if (!fs.is_directory(file) && file.extension() == ".cmake")
+                if (Strings::case_insensitive_ascii_equals(vcpkg::u8string(file.extension()), ".cmake"))
```

Comment:
I changed any kind of file name comparisons to use `case_insensitive_ascii_***`. We probably should consistently use that or not use that?

Reply:
I think that this should be centralized and either:
1. use icase only on win32 and osx
2. match case everywhere

Even on windows, there is the WSL filesystem to worry about and I'm told that some users use case-sensitive stuff on OSX as well (because they hate their programs working or something 😛).

Reply:
I disagree with (1) -- that policy has gotten git into hot water quite often. I'm not sure if (2) is implementable.

Reply:
I did a more principled audit of this. I made the sensitivity choice in postbuildlint such that the most errors would be reported, if that makes sense. So if we are filtering out cmake files, I did equality, if we are finding cmake files to check, I did case insensitive, etc.